### PR TITLE
Add simple binary for load testing

### LINKED ISSUE #1986: Load testing of Oak functions

### PR DESCRIPTION
**Pull request by tiziano88**
Refactor client into separate library and binary.

Fix #1986

# Checklist

- [ ] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by
        [Cloudbuild](/cloudbuild.yaml)
  - [ ] I have updated [documentation](/docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

File: oak_functions/load_test/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        oak_functions_client::Client::new(&URL.parse().context("Could not parse URL")?)
+            .await
+            .context("Could not create client")?;
+
+    let mut latencies_millis = Vec::<f64>::with_capacity(TOTAL_REQUESTS);
+
+    for i in 0..TOTAL_REQUESTS {
+        let start = Instant::now();
+        let response = client
+            .invoke(REQUEST)
```
Comment (rbehjati) [inline oak_functions/load_test/src/main.rs:38]: Not sure how repeatedly sending the same request affects the measurements. But I think it should be fine for now.

File: oak_functions/load_test/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            std::str::from_utf8(response.body().unwrap()).unwrap()
+        );
+    }
+    println!("-------- total --------");
+    println!("mean: {:4.0}ms", &latencies_millis.mean());
+    println!("min: {:4.0}ms", &latencies_millis.min());
+    println!("max: {:4.0}ms", &latencies_millis.max());
+    // See https://en.wikipedia.org/wiki/Median_absolute_deviation.
+    println!(
+        "median abs dev: {:4.0}ms",
```
Comment (rbehjati) [inline oak_functions/load_test/src/main.rs:57]:
```suggestion
        "median absolute deviation: {:4.0}ms",
```

File: oak_functions/client/src/bin/oak_functions_client.rs [hunk 3]
```diff
@@ -17,17 +17,10 @@
 //! Sends a gRPC request to the weather lookup application and checks that the response has the
 //! correct format.
 
-pub mod proto {
```
Comment (rbehjati) [inline oak_functions/client/src/bin/oak_functions_client.rs:20]: Why is this file moved? Wouldn't `oak_functions/client/src/main.rs` work?

**Review by rbehjati — APPROVED** [review]

File: oak_functions/client/src/bin/oak_functions_client.rs (hunk 3 above)
Comment (tiziano88) [reply oak_functions/client/src/bin/oak_functions_client.rs]: I thought I tried that first and it didn't work for some reason, but now trying again it actually works :thinking: moved back!

File: oak_functions/load_test/src/main.rs (hunk 1 above)
Comment (tiziano88) [reply oak_functions/load_test/src/main.rs:38]: We don't do any caching, so it should be pretty realistic (modulo internal CPU caches)

File: oak_functions/load_test/src/main.rs (hunk 2 above)
Comment (tiziano88) [reply oak_functions/load_test/src/main.rs]: Done.
